packages/core/src/moduleRecord.js

```diff
@@ -1,77 +1,38 @@
 // @ts-check
 
+/**
+ * Do not import this; use `import('@lavamoat/types').LavamoatModuleRecord`
```

Comment:
It's not deprecated.  So the point is that we need to use `LavamoatModuleRecord` from the `types` package as an interface and implement it.  

In JSDoc--and I think this is a bug--`@implements` works _if and only if_ the literal type you're using is a `@typedef` _in the current module_.  That means:

```js
/** @typedef {import('./butts').Butts} Butts */

/**
 * @implements {Butts}
 */
class Pants {
}
``` 

works fine, but:

```js
/**
 * @implements {import('./butts').Butts}
 */
class Pants {
}
``` 

_does not_.


But if we use a `@typedef`, we re-export the type, because that's what typedefs do.  But we only want one source of truth for the `LavamoatModuleRecord` interface.  Because we have to use the `@typedef`, we've now created a "copy" of `LavamoatModuleRecord` named `LMR`.  Nobody should use it.

That said, if we can upgrade TS, we can probably avoid the `@typedef` altogether because of the new `@import` tag.